R/timeDifferences.R

```diff
@@ -0,0 +1,35 @@
+#' Consecutive time difference of numeric values can be calculated
+#' as well as the time difference based on the certain time point with
+#' parameter `baseline`.
+#'
+#' @param time_field : vector containing time points
+#' @param baseline : integer, value to calculate the distance from certain time
+#' point. (default: \code{baseline = 1})
+#'
+#' @examples
+#' data(hitchip1006)
+#'
+#' diffHitchip <- timeDifferences(time_field = colData(hitchip1006)[,"time"],
+#'                                              baseline = 1)
+#'
+#' ##Each of these fields in the list object can be applied to the
+#' #\linkS4class{SummarizedExperiment} object as new fields.
+#'
+#' colData(hitchip1006)$time_diff <- diffHitchip$Consecutive
+#' colData(hitchip1006)$base_diff <- diffHitchip$Baseline
+#'
+#' @return a list object containing consecutive time differences and
+#' the distance from selected time point to each time point.
+#'
+#' @export
+timeDifferences <- function(time_field, baseline = 1){
```

Comment:
The baseline shift is covered already by lubridate (`x - x[[1]]`). This should be sufficient, with an example in OMA on how to add this field to colData.

The time difference is covered already by lubridate `diff(x)`. This should be sufficient, with an example in OMA on how to add this field to colData.

Conclusion: this function is not needed as such. 

However, we would like to do this per subject. I suggest to remove this function from the current PR and merge the other things when they are ready. Then in a separate PR, let us create a new function that can split samples per subject, add these informations per subject, and then collect the data back to the original colData. This will complicate the present PR too much, therefore let's remove this function from this PR.